### PR TITLE
redo(ticdc): owner and processores shouldn't share one redo log writer

### LINKED ISSUE #6695: redo log: change stuck in case redo_apply_multiple_workload_node_restart_sync
### What did you do?

- create changefeed1: enable redo log, filter sysbench table
- create changefeed2: enable redo log, filter go-ycsb table
- run sysbench prepare, meanwhile, all cdc node restart
- run truncate all sysbench table
- run sysbench run, meanwhile, all pd restart
- drop all sysbench tables
- run sysbench prepare again
- run go-ycsb workload
- sendfinishmark and waitforthisDDL

### What did you expect to see?

_No response_

### What did you see instead?

k0JvrUqnLJ

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):

```console
(paste TiCDC version here)
Release Version: v6.1.0-20220809
Git Commit Hash: 0c82c3405e49ac1da9f5e705d3862ec5a79c22bf
Git Branch: heads/refs/tags/v6.1.0-20220809
UTC Build Time: 2022-08-09 13:29:30
Go Version: go version go1.18.5 linux/amd64
Failpoint Build: false
```

### PR DESCRIPTION
**Pull request by hicqu**
Signed-off-by: qupeng <qupeng@pingcap.com>

### What problem does this PR solve?

Issue Number: close #6695 

### What is changed and how it works?

Reusing log write in redo manager should be in a correct way

#### Tests 

 - Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/redo/manager.go
Comment (nongfushanquan) [inline cdc/redo/manager.go:546]: too many logs will appear in the log files

**Review by nongfushanquan — COMMENTED** [review]
please and an issue

File: cdc/redo/manager.go
Comment (hicqu) [reply cdc/redo/manager.go:570]: There will be one this log per changefeed. So I guess it's acceptable.

File: cdc/redo/manager.go
Comment (liuzix) [inline cdc/redo/manager.go:371]: When will `ckpt <= curr` happen?

File: cdc/redo/manager.go
Comment (liuzix) [inline cdc/redo/manager.go:625]: I think this is a deadlock risk.

File: cdc/owner/changefeed.go
Comment (3AceShowHand) [inline cdc/owner/changefeed.go:494]:
```suggestion
zap.String("namespace", c.id.Namespace),
				zap.String("changefeed", c.id.ID))
```

File: cdc/processor/processor.go
Comment (3AceShowHand) [inline cdc/processor/processor.go:707]:
```suggestion
		zap.String("namespace", p.changefeedID.Namespace,
		zap.String("changefeed", p.changefeedID.ID)
```

File: cdc/redo/manager.go
Comment (3AceShowHand) [inline cdc/redo/manager.go:569]: please add a comment about your consideration here, why a buffered channel for error ?

File: cdc/redo/manager.go
Comment (hicqu) [reply cdc/redo/manager.go:286]: There is no explicit API to close `ManagerImpl`. Now background goroutines are controled with `context.Context`.

File: cdc/redo/manager.go
Comment (hicqu) [reply cdc/redo/manager.go:371]: It happends when the function is called in multiple threads.

File: cdc/redo/manager.go
Comment (hicqu) [reply cdc/redo/manager.go:625]: Owner and processor will cancel `ctx` as expected so here the `select` can finish finally.

File: cdc/processor/pipeline/sink.go
Comment (nongfushanquan) [inline cdc/processor/pipeline/sink.go:162]: why not merge this if condition into the next one

File: cdc/processor/pipeline/sink.go
Comment (nongfushanquan) [inline cdc/processor/pipeline/sink.go:212]: //todo : fix me, replace the checkpoint with the global checkpoint

File: cdc/redo/manager.go
Comment (nongfushanquan) [inline cdc/redo/manager.go:590]: Pls add log about the err

File: cdc/owner/changefeed.go
Comment (3AceShowHand) [inline cdc/owner/changefeed.go:494]:
```suggestion
		zap.String("changefeed", c.id.ID))
```

File: cdc/processor/processor.go
Comment (3AceShowHand) [inline cdc/processor/processor.go:713]:
```suggestion
		zap.String("changefeed", p.changefeedID.ID))
```

File: cdc/redo/manager.go
Comment (nongfushanquan) [reply cdc/redo/manager.go:371]: if global checkpoint is set here,, this condition will not happen?

File: cdc/redo/manager.go
Comment (zhaoxinyu) [inline cdc/redo/manager.go:646]:
```suggestion
			log.Info("redo manager GC exits as context cancelled",
```

File: cdc/redo/options.go
Comment (zhaoxinyu) [inline cdc/redo/options.go:47]:
```suggestion
// NewProcessorManagerOptions creates a manager options for processor.
```

File: cdc/redo/options.go
Comment (zhaoxinyu) [inline cdc/redo/options.go:58]:
```suggestion
// newMockManagerOptions creates a manager options for mock tests.
```

File: cdc/redo/writer/writer.go
Comment (zhaoxinyu) [inline cdc/redo/writer/writer.go:454]:
```suggestion
	var rowStopped, ddlStopped bool
```

File: cdc/redo/writer/writer.go
Comment (zhaoxinyu) [inline cdc/redo/writer/writer.go:459]:
```suggestion
		ddlStopped = !l.ddlWriter.IsRunning()
```

File: cdc/redo/writer/writer.go
Comment (zhaoxinyu) [inline cdc/redo/writer/writer.go:461]:
```suggestion
	return rowStopped || ddlStopped
```

**Review by zhaoxinyu — APPROVED** [review]

**Review by 3AceShowHand — APPROVED** [review]

**Pull request merged**
